File: atcoder/abc/4/44/d.py
N = int(input())
A = [int(Ai) for Ai in input().split()]
A.sort()

ans = [0]
for i in range(A[0]):
    ans[0] = ans[0] + 10**i
(x, y) = divmod(ans[0] * N, 10**A[0])
ans[0] = y
ans.append(x)

for i in range(1, N):
    if A[i] == A[i-1]:
        continue

    bi = 0
    for j in range(A[i]-A[i-1]):
        bi = bi + 10**j

    (x, y) = divmod(bi * (N-i) + ans[-1], 10**(A[i]-A[i-1]))
    ans[-1] = y
    ans.append(x)

if ans[-1] == 0 and len(ans) > 1:
    ans.pop()
for i in range(0, len(ans)):
    print(ans[len(ans)-1-i], end='')

File: atcoder/abc/4/44/d.cpp
#include <bits/stdc++.h>
using namespace std;

#define divmod(n, d) make_pair((n) / (d), (n) % (d))

int main() {
	uint N;
	cin >> N;
	vector<uint> A(N);
	for (uint& Ai : A) cin >> Ai;
	ranges::sort(A);

	vector ans = {0ul};
	for (uint i = 0; i < N; i++) {
		if (i > 0 && A[i] == A[i - 1]) continue;
		for (uint j = ans.size(); j <= A[i]; j++) {
			const auto& [x, y] = divmod(ans.back() + N - i, uint64_t(10));
			ans.back()         = y;
			ans.push_back(x);  // 最大 6e15 くらい？
		}
	}

	if (ans.size() > 1 && ans.back() == 0) ans.pop_back();
	for_each(ans.rbegin(), ans.rend(), [](auto ci) { cout << ci; });

	return 0;
}
